### DISCUSSION TITLE
Vite + polyfills: globalThis is not defined

### DISCUSSION THREAD

**Discussion by luzzardik** [nuxt/nuxt#18900] [Questions]
Hi everyone,

I'm trying to make an app using Nuxt 3 on an Android device based on Android 8.1.
I need to add Vite polyfills, but it doesn't seem to work.

Here's my config:
```vite: {
        plugins: [
            legacy({
                targets: ['chrome >= 71', 'edge >= 79', 'safari >= 11.1', 'firefox >= 67'],
                //ignoreBrowserslistConfig: true,
                renderLegacyChunks: false,
                modernPolyfills: true,
            }),
        ],
    }
```

Does anyone has a clue ?

**Comment by rafifn** [comment]
I have fixed it in my project in the following way

```
//app.vue

useHead({
  script: [{ children: 'if (globalThis === undefined) var globalThis = window;' }],
})
```
but, I have another issue after that...
the user cannot do anything on the client side.

currently, `@vitejs/plugin-legacy` can not be used in nuxt 3

https://github.com/nuxt/nuxt/issues/15464
https://github.com/nuxt/nuxt/issues/14634

**Reply by luzzardik** [reply]
Yeah for now I'm just using plain old Vite without Nuxt 3; I'll switch back when it works
I have to use f*d up devices based on PayDroid 8
There is a lot of missing polyfills

**Reply by rafifn** [reply]
@danielroe It seems that many are waiting for compatibility improvements with older browsers 😁
